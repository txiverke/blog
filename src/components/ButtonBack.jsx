// @flow

import React from 'react'
import { Link } from 'react-router-dom'

type Props = {
  label: string,
  route: string,
}

const ButtonBack = ({ label, route }: Props) => 
  <Link
    to={route} 
    className="app-header-nav-item btn btn-icon btn-back txt-tiny tr">
    <span className={`mr5 icon-cheveron-left`}></span>
    <span className="btn-back-txt">{label}</span>
  </Link>

ButtonBack.defaultProps = {
  label: 'Back',
  route: '/'
}

export default ButtonBack