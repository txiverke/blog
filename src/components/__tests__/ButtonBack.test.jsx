// @flow

import React from 'react'
import { shallow } from 'enzyme'
import ButtonBack from '../ButtonBack'

describe('<ButtonBack />', () => {
  it('should render itself', () => {
    const component = shallow(<ButtonBack />)
    expect(component).toMatchSnapshot()
  })

  it('should render the optional props', () => {
    const label = 'test label'
    const route = '/test'
    const component = shallow(<ButtonBack label={label} route={route} />)
    expect(component).toMatchSnapshot()
  })
})