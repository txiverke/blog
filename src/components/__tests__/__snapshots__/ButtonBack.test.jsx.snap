// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ButtonBack /> should render itself 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ButtonBack
    label="Back"
    route="/"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <span
          className="mr5 icon-cheveron-left"
/>,
        <span
          className="btn-back-txt"
>
          Back
</span>,
      ],
      "className": "app-header-nav-item btn btn-icon btn-back txt-tiny tr",
      "replace": false,
      "to": "/",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "mr5 icon-cheveron-left",
        },
        "ref": null,
        "rendered": null,
        "type": "span",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Back",
          "className": "btn-back-txt",
        },
        "ref": null,
        "rendered": "Back",
        "type": "span",
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <span
            className="mr5 icon-cheveron-left"
/>,
          <span
            className="btn-back-txt"
>
            Back
</span>,
        ],
        "className": "app-header-nav-item btn btn-icon btn-back txt-tiny tr",
        "replace": false,
        "to": "/",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "mr5 icon-cheveron-left",
          },
          "ref": null,
          "rendered": null,
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Back",
            "className": "btn-back-txt",
          },
          "ref": null,
          "rendered": "Back",
          "type": "span",
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`<ButtonBack /> should render the optional props 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ButtonBack
    label="test label"
    route="/test"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <span
          className="mr5 icon-cheveron-left"
/>,
        <span
          className="btn-back-txt"
>
          test label
</span>,
      ],
      "className": "app-header-nav-item btn btn-icon btn-back txt-tiny tr",
      "replace": false,
      "to": "/test",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "mr5 icon-cheveron-left",
        },
        "ref": null,
        "rendered": null,
        "type": "span",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "test label",
          "className": "btn-back-txt",
        },
        "ref": null,
        "rendered": "test label",
        "type": "span",
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <span
            className="mr5 icon-cheveron-left"
/>,
          <span
            className="btn-back-txt"
>
            test label
</span>,
        ],
        "className": "app-header-nav-item btn btn-icon btn-back txt-tiny tr",
        "replace": false,
        "to": "/test",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "mr5 icon-cheveron-left",
          },
          "ref": null,
          "rendered": null,
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "test label",
            "className": "btn-back-txt",
          },
          "ref": null,
          "rendered": "test label",
          "type": "span",
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
